### PR TITLE
Add migration for Python 3.12

### PR DESCRIPTION
**Pull request by xhochy**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/migrations/python312.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - pypy3.6
+      - pypy-meta
+      - cross-python
+      - python_abi
+    exclude_pinned_pkgs: false
+
+python:
+  - 3.12.* *_cpython
+channel_sources:
+  - conda-forge/label/python_rc,conda-forge
```
Comment (xhochy) [inline recipe/migrations/python312.yaml:35]: As one can see in https://github.com/conda-forge/pydantic-core-feedstock/pull/63/files, this line has sadly no effect. I would be happy for support on how to achieve this.

**Review by ocefpaf — APPROVED** [review]

File: recipe/migrations/python312.yaml (hunk 1 above)
Comment (xhochy) [reply recipe/migrations/python312.yaml:35]: This needs a new release of `conda-smithy`, then we're good to go.

File: recipe/migrations/python312.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    pr_limit: 20
+    max_solver_attempts: 3  # this will make the bot retry "not solvable" stuff 12 times
+    exclude:
+      # this shouldn't attempt to modify the python feedstocks
+      - python
+      - pypy3.6
+      - pypy-meta
+      - cross-python
+      - python_abi
+    exclude_pinned_pkgs: false
```
Comment (0xbe7a) [inline recipe/migrations/python312.yaml:30]:
```suggestion
    exclude_pinned_pkgs: false
    additional_zip_keys:
      - channel_sources
```

**Comment by hmaarrfk** [comment]
great work.

File: recipe/migrations/python312.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    additional_zip_keys:
+      - channel_sources
```
Comment (isuruf) [inline recipe/migrations/python312.yaml:32]: See https://github.com/conda-forge/conda-smithy/pull/1764#issuecomment-1726250148

**Review by isuruf — CHANGES_REQUESTED** [review]

File: recipe/migrations/python312.yaml (hunk 3 above)
Comment (xhochy) [reply recipe/migrations/python312.yaml:32]: I would release conda-smithy 3.26.2 later and would wait for a merge of https://github.com/conda-forge/cross-python-feedstock/pull/74. Are then all concerns regarding the migration addressed?

Comment (isuruf) [reply recipe/migrations/python312.yaml:32]: Yes

**Pull request merged**